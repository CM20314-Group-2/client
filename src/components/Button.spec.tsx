import Button from '@components/Button'
import { fireEvent, render, waitFor } from '@testing-library/react-native'

describe('Button', () => {
  it('fires onPress function when pressed', async () => {
    const mockOnPress = jest.fn()
    const {getByTestId, toJSON} = render(<Button text='asd' onPress={mockOnPress}/>)

    const button = getByTestId('button')
    fireEvent.press(button)
    await waitFor(() => {
      expect(mockOnPress).toHaveBeenCalled()
    })

    expect(toJSON()).toMatchSnapshot()
  })
})