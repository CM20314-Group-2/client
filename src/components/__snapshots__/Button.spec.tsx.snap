// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#1ea853",
        "borderRadius": 5,
        "justifyContent": "center",
        "width": "75%",
      },
    ]
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "bold",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    asd
  </Text>
</View>
`;
